src/emqtt_sock.erl

```diff
@@ -147,3 +148,15 @@ apply_sni(Opts, Host) ->
         _ ->
             Opts
     end.
+
+apply_host_check_fun(Opts) ->
+    case proplists:is_defined(customize_hostname_check, Opts) of
+        true ->
+            Opts;
+        false ->
+            %% Default Support wildcard cert
+            DefHostCheck = {customize_hostname_check,
+                            [{match_fun,
+                              public_key:pkix_verify_hostname_match_fun(https)}]},
```

Comment:
does this work with usual hostnames ?
should perhaps add a test case for regression purposes.

Reply:
tests are covered here:
https://github.com/emqx/emqtt/pull/179/commits/f572b8455625afdd4e59ea3a57767489e91cca89